controllers/component_build_controller.go

```diff
@@ -286,67 +287,123 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 
 	// Check if Pipelines as Code workflow enabled
-	if val, exists := component.Annotations[PaCProvisionAnnotationName]; exists {
-		if val != PaCProvisionRequestedAnnotationValue && !isSwitchedImageRegistry {
-			if !(val == PaCProvisionDoneAnnotationValue || val == PaCProvisionErrorAnnotationValue) {
-				message := fmt.Sprintf(
-					"Unexpected value \"%s\" for \"%s\" annotation. Use \"%s\" value to do Pipeline as Code provision for the Component",
-					val, PaCProvisionAnnotationName, PaCProvisionRequestedAnnotationValue)
-				log.Info(message)
-			}
-			// Nothing to do
-			return ctrl.Result{}, nil
+	if pacAnnotationValue, exists := component.Annotations[PaCProvisionAnnotationName]; exists {
+		// Always create new / update existing PaC configuration PR if image registry has changed.
```

Comment:
The code actually implies the meaning this comment tells. It'll be good to tell why doing the PaC workflow provision workflow when switch to image repository generated by image-controller.